R/read-gs.R

```diff
@@ -15,11 +16,200 @@ read_gs <- function(path, verbose = FALSE){
   gs <- read_csv(path, trim_ws = FALSE) |>
     #check format
     check_data_format()
+  #add source attribute
+  attr(gs, "source") <- "Gradescope"
+  
+  gs
+}
+
+
+#'Read in 1 or more files from Grading Platform
+#'
+#'This function will read in input data and combine in the appropriate manner.
+#'@param grades_path Is the path to the csv containing graded assignments.
+#'@param other_file_paths Is a list containing any other filepaths desired to be read in. Each name should be a designated type of file. For example, "lateness" and "roster" and the value should be the filepath.
+#'@param source Determines how to read the grade file. By default, it is auto so that Gradebook can determine what format the file is in. Alternatively, users can specify "Gradescope" or "Canvas" to force certain interpretations.
+#'@return A dataframe of grades. Later support for other files will be returned as a list as well.
+#'@importFrom readr read_csv
+#'@export
+read_files <- function(grades_path, 
+                       other_file_paths = list(), 
+                       source = "auto"){
+  #first read in all of the files
+  
+  #begininng with the grades csv
+  
+  tryCatch({
+    grades <- readr::read_csv(grades_path, trim_ws = FALSE)
+    },
+    error = function(e){
+      stop(
+      paste0("Error when attempting to read grade csv file. 
+           Check that the filepath is correct. Error: ",
+             e)
+      )
+    }
+  )
+  
+  #now read in other supplied data 
+ for (data_type in names(other_file_paths)){
+   warning(paste0(data_type, 
+                  " is not currently supported as an extra file source."))
+ }
+  
+  # now determine how to read in data
+  
+  auto_determine <- determine_grade_source(grades)
+  
+  if (source == "auto"){
+    #fill in later
+    
+    if (auto_determine == "Gradescope"){
+      attr(grades, "source") <- "Gradescope"
+      
+      return(check_data_format(grades))
+    } else if (auto_determine == "Canvas"){
+      #the read_canvas_grades does all of the lifting
+      return(read_canvas_grades(grades))
```

Comment:
consider pulling `check_data_format()` out of the end of `read_canvas_grades()` and putting it here to mirror the gradescope file (or make identity read fn for gradescope and put it there).

Reply:
Added near identity function for Gradescope!